src/pyscipopt/scip.pyx

```diff
@@ -547,6 +649,55 @@ cdef class Node:
         """Is the node marked to be propagated again?"""
         return SCIPnodeIsPropagatedAgain(self.scip_node)
 
+    def getNParentBranchings(self):
+        """Retrieve the number of variable branchings that were performed in the parent node to create this node."""
+        cdef SCIP_VAR* dummy_branchvars
+        cdef SCIP_Real dummy_branchbounds
+        cdef SCIP_BOUNDTYPE dummy_boundtypes
+        cdef int nbranchvars
+        # This is a hack: the SCIP interface has no function to directly get the
+        # number of parent branchings, i.e., SCIPnodeGetNParentBranchings() does
+        # not exist.
+        SCIPnodeGetParentBranchings(self.scip_node, &dummy_branchvars,
+                                    &dummy_branchbounds, &dummy_boundtypes,
+                                    &nbranchvars, 0)
+        return nbranchvars
+
+    def getParentBranchings(self):
+        """Retrieve the set of variable branchings that were performed in the parent node to create this node."""
+        cdef int nbranchvars = self.getNParentBranchings()
+        if nbranchvars == 0:
+            return []
+
+        cdef SCIP_VAR** branchvars = <SCIP_VAR**> malloc(nbranchvars * sizeof(SCIP_VAR*))
+        cdef SCIP_Real* branchbounds = <SCIP_Real*> malloc(nbranchvars * sizeof(SCIP_Real))
+        cdef SCIP_BOUNDTYPE* boundtypes = <SCIP_BOUNDTYPE*> malloc(nbranchvars * sizeof(SCIP_BOUNDTYPE))
+
+        SCIPnodeGetParentBranchings(self.scip_node, branchvars, branchbounds,
+                                    boundtypes, &nbranchvars, nbranchvars)
+
+        branchings = [Branching.create(branchvars[i], branchbounds[i], boundtypes[i])
+                      for i in range(nbranchvars)]
+
+        free(boundtypes)
+        free(branchbounds)
+        free(branchvars)
+        return branchings
```

Comment:
I guess the discussion about the branching class is going to be relevant here. So, in principle, this function should just return a tuple of variables, bounds and boundtype.

Reply:
This is indeed how I implemented this originally, but it was a bit clunky to use because (a) the return values are arranged in an unnatural way (by variables, bounds, boundtype rather than by branching), and (b) printing the return values was not helpful. (In the current form, printing out the return value gives a list of inequalities of the form x_i <= u_i, etc.)

Reply:
I've changed the method so that it returns a tuple of variables, bounds, boundtypes.